programmers-coding-test/kimjiwon/binary-search/떡볶이떡만들기.js

```diff
@@ -0,0 +1,21 @@
+const line = require("fs").readFileSync("/dev/stdin", "utf8");
+let input = line.trim().split("\n");
+
+const [N, M] = input[0].split(" ").map(Number);
+const riceCake = input[1]
+  .split(" ")
+  .map(Number)
+  .sort((a, b) => b - a);
+let start = 0;
+let end = riceCake[0];
+let result = 0;
+
+while (start <= end) {
+  let mid = parseInt((start + end) / 2);
+  let client = riceCake
+    .map((x) => x - mid)
+    .filter((num) => num > 0)
+    .reduce((acc, x) => acc + x);
```

Comment:
while문이 돌 때마다 그 안에서 반복문을 세번이나 더 도네!!
효율적이지 못해 ㅋㅋㅋㅋ

아래처럼하면 reduce 하나만으로도 구현이 가능하겠다! (난 이렇게 풀진 않았음)
그리고 client는 밑에서 변경되지 않으므로 let이 아닌 const가 맞음. (이거 언제까지 실수할거니)
```suggestion
  const client = riceCake
    .reduce((acc, x) => {
        if (x > mid) acc + x;
        return acc;
    }, 0);
```